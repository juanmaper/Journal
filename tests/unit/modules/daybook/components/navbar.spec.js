import { shallowMount } from '@vue/test-utils'

import NavBar from '@/modules/daybook/components/NavBar.vue'
import createVuexStore from '../../../mocks-data/mock-store'


describe('Navbar component tests', () => {

    const store = createVuexStore({
        user: {
            name: 'Juanma',
            email: 'juanma@gmail.com'
        },
        status: 'authenticated',
        idToken: 'ABC',
        refreshToken: 'XYZ'
    })

    beforeEach(() =>  jest.clearAllMocks() )


    test('should properly show the component', () => {
        
        const wrapper = shallowMount( NavBar, {
            global: {
                plugins: [ store ]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()

    })

    
    
    
    
})