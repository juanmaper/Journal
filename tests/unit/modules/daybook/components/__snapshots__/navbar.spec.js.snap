// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component tests should properly show the component 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white"><img src="" alt="Vue logo" height="24" class="d-inline-block align-text-top mx-2"> </a>
  <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
